# Earthquake and FAANG Data Analysis

### This notebook analyzes earthquake and stock market datasets using pandas and NumPy.

# PART-1

Load the dataset data/parsed.csv into a pandas DataFrame and complete the
following tasks using only pandas and NumPy where necessary.

In [78]:
#Import libraries
import pandas as pd
import numpy as np

In [7]:
#Load the dataset
df = pd.read_csv("/home/marie/code/Python-AI_assignment/assign3_data/parsed.csv")

In [32]:
df.tail()

,alert,cdi,code,detail,dmin,felt,gap,ids,mag,magType,...,status,time,title,tsunami,type,types,tz,updated,url,parsed_place
9327,NaN,NaN,73086771,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.01806,NaN,185.0,",nc73086771,",0.62,md,...,reviewed,1537230228060,"M 0.6 - 9km ENE of Mammoth Lakes, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.0,1537285598315,https://earthquake.usgs.gov/earthquakes/eventp...,California
9328,NaN,NaN,38063967,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.03041,NaN,50.0,",ci38063967,",1.00,ml,...,reviewed,1537230135130,"M 1.0 - 3km W of Julian, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,scit...",-480.0,1537276800970,https://earthquake.usgs.gov/earthquakes/eventp...,California
9329,NaN,NaN,2018261000,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.45260,NaN,276.0,",pr2018261000,",2.40,md,...,reviewed,1537229908180,"M 2.4 - 35km NNE of Hatillo, Puerto Rico",0,earthquake,",geoserve,origin,phase-data,",-240.0,1537243777410,https://earthquake.usgs.gov/earthquakes/eventp...,Puerto Rico
9330,NaN,NaN,38063959,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.01865,NaN,61.0,",ci38063959,",1.10,ml,...,reviewed,1537229545350,"M 1.1 - 9km NE of Aguanga, CA",0,earthquake,",focal-mechanism,geoserve,nearby-cities,origin...",-480.0,1537230211640,https://earthquake.usgs.gov/earthquakes/eventp...,California
9331,NaN,NaN,38063935,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.01698,NaN,39.0,",ci38063935,",0.66,ml,...,reviewed,1537228864470,"M 0.7 - 9km NE of Aguanga, CA",0,earthquake,",focal-mechanism,geoserve,nearby-cities,origin...",-480.0,1537305830770,https://earthquake.usgs.gov/earthquakes/eventp...,California


In [26]:
df.columns

Index(['alert', 'cdi', 'code', 'detail', 'dmin', 'felt', 'gap', 'ids', 'mag',
       'magType', 'mmi', 'net', 'nst', 'place', 'rms', 'sig', 'sources',
       'status', 'time', 'title', 'tsunami', 'type', 'types', 'tz', 'updated',
       'url', 'parsed_place'],
      dtype='object')

#### 1. Percentile Calculation: 
Find the 95th percentile of earthquake magnitude in Japan where the magType is 'mb'.

In [39]:
japan_mb = df[(df['parsed_place'].str.contains("Japan", case= False, na=False)) & (df['magType'] == 'mb')]

percentile_95 = np.percentile(japan_mb['mag'], 95)


In [49]:
print("95Th percentile magnitude in Japan (mb):", percentile_95)

95Th percentile magnitude in Japan (mb): 4.9


#### 2. Tsunami Percentage (Indonesia): 
Calculate the percentage of earthquakes in Indonesia that were associated with a tsunami.

In [52]:
indonesia_eq = df[df['parsed_place'].str.contains("Indonesia", case= False, na= False)]

tsunami_count = indonesia_eq['tsunami'].sum()
total_eq = len(indonesia_eq)

tsunami_percentage = (tsunami_count/total_eq) * 100

In [53]:
print("Tsunami percentage in Indonesia:", tsunami_percentage)

Tsunami percentage in Indonesia: 23.12925170068027


#### 3. Summary Statistics (Nevada): 
Generate summary statistics (count, mean, std, min, quartiles, max) for earthquakes that occurred in Nevada.

In [59]:
nevada_eq = df[df['parsed_place'].str.contains("Nevada", case = False, na = False)]

summary_stats = nevada_eq['mag'].describe()

In [60]:
print(summary_stats)

count    681.000000
mean       0.500073
std        0.696710
min       -0.500000
25%       -0.100000
50%        0.400000
75%        0.900000
max        2.900000
Name: mag, dtype: float64


#### 4. Ring of Fire Classification:
Add a new column to the DataFrame called ring_of_fire that indicates whether the earthquake occurred in a country or U.S. state located on the
Ring of Fire.

Use the following locations:

Bolivia, Chile, Ecuador, Peru, Costa Rica, Guatemala, Mexico, *(Be careful not to include New Mexico)* Japan, Philippines, Indonesia, New Zealand, Antarctic
Canada, Fiji, Alaska, Washington, California
Russia, Taiwan, Tonga, Kermadec Islands

The column should contain:

● True if the earthquake occurred in one of these locations

● False otherwise

In [61]:
ring_locations = {"Bolivia", "Chile", "Ecuador", "Peru", "Costa Rica", "Guatemala", "Mexico", "Japan", "Philippines", "Indonesia", "New Zealand", "Antarctic", "Canada", "Fiji", "Alaska", "Washington", "California", "Russia", "Taiwan", "Tonga", "Kermadec Islands"}

In [62]:
def check_ring_of_fire(parsed_place):
    if pd.isna(parsed_place):
        return False
    
    for loc in ring_locations:
        if loc in parsed_place:
            if "New Mexico" in parsed_place:
                return False
            return True
    
    return False

In [66]:
df['ring_of_fire'] = df['parsed_place'].apply(check_ring_of_fire)

print(df[['parsed_place', 'ring_of_fire']].tail())

     parsed_place  ring_of_fire
9327   California          True
9328   California          True
9329  Puerto Rico         False
9330   California          True
9331   California          True


#### 5. Earthquake Counts (Inside vs Outside Ring of Fire)
Calculate:

● The number of earthquakes that occurred within the Ring of Fire

● The number of earthquakes that occurred outside the Ring of Fire

In [69]:
ring_counts = df['ring_of_fire'].value_counts()

inside_ring = ring_counts.get(True, 0)
outside_ring = ring_counts.get(False, 0)

In [70]:
print("Earthquake occured within the Ring of fire:", inside_ring)
print("Earthquake occured outside the Ring of fire:", outside_ring)


Earthquake occured within the Ring of fire: 7188
Earthquake occured outside the Ring of fire: 2144


#### 6. Tsunami Count Along the Ring of Fire
Find the total number of earthquakes in Ring of Fire locations that were associated with a
tsunami.

In [74]:
ring_tsunami = df[(df['ring_of_fire'] == True) & (df['tsunami']==1)]

tsunami_ring_counts = len(ring_tsunami)

In [75]:
print("Tsunami earthquakes in Ring of fire:", tsunami_ring_counts)

Tsunami earthquakes in Ring of fire: 45


# PART-II
Using the CSV files in the data/ folder and what we have learned so far, complete the
following exercises:

In [76]:
#import libraries (as here's already imported in above part)

# import pandas as pd
# import numpy as np

In [80]:
#Load dataset from the data folder

earthquakes = pd.read_csv('assign3_data/earthquakes.csv')
faang = pd.read_csv('assign3_data/faang.csv')
parsed = pd.read_csv('assign3_data/parsed.csv')

In [84]:
# earthquakes.head(3)
faang.head(3)
# parsed.tail(3)

,ticker,date,high,low,open,close,volume
0,FB,2018-01-02,181.580002,177.550003,177.679993,181.419998,18151900.0
1,FB,2018-01-03,184.779999,181.330002,181.880005,184.669998,16886600.0
2,FB,2018-01-04,186.210007,184.100006,184.899994,184.330002,13880900.0


#### 1. Filtering (earthquakes.csv):
Select all earthquakes that occurred in Japan with magType equal to 'mb' and magnitude ≥ 4.9.